pkg/proxy/winkernel/proxier.go

```diff
@@ -1149,8 +1149,27 @@ func (proxier *Proxier) syncProxyRules() {
 			klog.V(3).Infof("Hns LoadBalancer resource created for nodePort resources %v, Id [%s]", svcInfo.clusterIP, hnsLoadBalancer.hnsID)
 		}
 
+		if svcInfo.onlyNodeLocalEndpoints {
+			// Only apply externalIPs policy for local endpoints.
+			localEndpoints := []endpointsInfo{}
```

Comment:
Instead of this, can you maintain a separate list of local endpoints. See [Line 1109 of PR 74825](https://github.com/kubernetes/kubernetes/pull/74825/commits/4e0f63b688891df26b1e138a329c2096051871b8#diff-2945fb41a0aa16fb820ff19341875989R1109)

We want to be able to have access to a list of all endpoints and a list of local endpoints for different DSR scenarios. Also, this approach is less verbose.

Then if svcInfo.onlyNodeLocalEndpoints is true, then you can just do hnsEndpoints = localEndpoints